src/util/triplea/MapXMLCreator/MapXMLCreator.java

```diff
@@ -0,0 +1,839 @@
+				break;
+			
+			case 16:
+				layoutNotesPanel();
+
+		default:
+			break;
+		}
+		final JFrame me = this;
+		SwingUtilities.invokeLater(new Runnable()
+		{
+			public void run()
+			{
+				me.pack();
+				m_stepActionPanel.revalidate();
+				m_stepActionPanel.repaint();
+			}
+		});
+	}
+	
+	protected void layoutNotesPanel()
+	{
+		m_stepActionPanel.setLayout(new GridBagLayout());
+		
+		final JTextArea taNotes = new JTextArea(MapXMLHelper.s_notes);
+		taNotes.addFocusListener(new FocusListener()
+		{
+			
+			@Override
+			public void focusLost(FocusEvent arg0)
+			{
+				if (!MapXMLHelper.s_notes.equals(taNotes.getText()))
+				{
+					MapXMLHelper.s_notes = taNotes.getText().trim();
+				}
+			}
+			
+			@Override
+			public void focusGained(FocusEvent arg0)
+			{
+			}
+		});
+		final JScrollPane spTaNotes = new JScrollPane(taNotes);
+		
+		Dimension size = m_stepActionPanel.getSize();
+		size.height -= 100;
+		spTaNotes.setMinimumSize(size);
+		
+		GridBagConstraints gbc_lFirstRow = new GridBagConstraints();
+		gbc_lFirstRow.insets = new Insets(0, 0, 5, 0);
+		gbc_lFirstRow.gridy = 0;
+		gbc_lFirstRow.gridx = 0;
+		gbc_lFirstRow.anchor = GridBagConstraints.NORTH;
+		gbc_lFirstRow.weightx = 1.0;
+		gbc_lFirstRow.weighty = 1.0;
+		gbc_lFirstRow.gridwidth = 3;
+		m_stepActionPanel.add(spTaNotes, gbc_lFirstRow);
+
+		GridBagConstraints gbc_lSecondRow = (GridBagConstraints) gbc_lFirstRow.clone();
+		gbc_lSecondRow.weighty = 0.0;
+		gbc_lSecondRow.insets = new Insets(0, 0, 0, 0);
+		gbc_lSecondRow.gridwidth = 1;
+		gbc_lSecondRow.gridx = 1;
+		gbc_lSecondRow.gridy = 1;
+		
+		final JButton bPreviewHTML = new JButton("Preview HTML");
+		bPreviewHTML.setPreferredSize(new Dimension(300, 30));
+		bPreviewHTML.setAction(new AbstractAction("Preview HTML")
+		{
+			private static final long serialVersionUID = -5608941822299486808L;
+			
+			public void actionPerformed(final ActionEvent event)
+			{
+				showHTML(MapXMLHelper.s_notes, "HTML Preview");
+			}
+		});
+		m_stepActionPanel.add(bPreviewHTML, gbc_lSecondRow);
+
+		GridBagConstraints gbc_lThirdRow = (GridBagConstraints) gbc_lSecondRow.clone();
+		gbc_lThirdRow.gridy = 2;
+		m_stepActionPanel.add(new JLabel("<html><big>Congratulation!</big></html>"), gbc_lThirdRow);
+		
+		GridBagConstraints gbc_lFourthRow = (GridBagConstraints) gbc_lSecondRow.clone();
+		gbc_lFourthRow.gridy = 3;
+		m_stepActionPanel.add(new JLabel("<html><nobr>You have completed all the steps for creating the map XML.</nobr></html>"), gbc_lFourthRow);
+		
+		final JButton bSave = new JButton("Save Entries to XML");
+		bSave.setPreferredSize(new Dimension(600, 35));
+		bSave.setAction(new AbstractAction("Save Map XML")
+		{
+			private static final long serialVersionUID = -5608941822299486808L;
+			
+			public void actionPerformed(final ActionEvent event)
+			{
+				MapXMLHelper.saveXML();
+			}
+		});
+		GridBagConstraints gbc_lFifthRow = (GridBagConstraints) gbc_lSecondRow.clone();

```

Comment:
We decided pretty early that google java style would be the arbitrer of any formatting/styling differences: https://google.github.io/styleguide/javaguide.html#s5.2.7-local-variable-names

We have one exception with line length where we do not follow the guide.

So, the name is going to be pretty long:
`gridBagConstraintsFifthRow`

I do not know what the "l" means in this name: "gbc_lFifthRow"

IMO a better name would be "fifthRowGridBag". It would be even better to not encode "fifth" in the variable name. If this class is modified and a row is inserted, then this is a completely unrelated line that now needs to be updated (an example of unnecessary coupling)